Formula/infer.rb

```diff
@@ -70,6 +70,10 @@ def install
     # Release build
     touch ".release"
 
+    # Disable handling external dependencies as opam is not aware of Homebrew on Linux.
+    # Error:  Package conflict!  * Missing dependency:  - conf-autoconf
+    on_linux { inreplace "build-infer.sh", "infer \"$INFER_ROOT\" $locked", "\\0 --no-depexts" }
```

Comment:
We should upstream a patch that allows us to pass this flag.

Reply:
Will try to work on separate upstream enhancement.

Upstream `master` build script has modified this specific line, so we need different fixes. If I remember correctly, `inreplace` should fail when it is unable to find pattern, so this should fail on next version bump.

Maybe just something like following:
```diff
diff --git a/build-infer.sh b/build-infer.sh
index 757d40293..2dd9a3076 100755
--- a/build-infer.sh
+++ b/build-infer.sh
@@ -21,6 +21,7 @@ INFER_OPAM_DEFAULT_SWITCH="4.12.0+flambda"
 INFER_OPAM_DEFAULT_SWITCH_OPTIONS="--package=ocaml-variants.4.12.0+options,ocaml-option-flambda"
 INFER_OPAM_SWITCH=${INFER_OPAM_SWITCH:-$INFER_OPAM_DEFAULT_SWITCH}
 INFER_OPAM_SWITCH_OPTIONS=${INFER_OPAM_SWITCH_OPTIONS:-$INFER_OPAM_DEFAULT_SWITCH_OPTIONS}
+INFER_OPAM_INSTALL_DEPS_OPTIONS=${INFER_OPAM_INSTALL_DEPS_OPTIONS:-""}
 PLUGIN_DIR="$INFER_ROOT/facebook-clang-plugins"
 PLUGIN_SETUP_SCRIPT=${PLUGIN_SETUP_SCRIPT:-setup.sh}
 PLUGIN_SETUP="${PLUGIN_DIR}/clang/${PLUGIN_SETUP_SCRIPT}"
@@ -145,7 +146,7 @@ install_opam_deps () {
     if [ "$USE_OPAM_LOCK" == yes ]; then
         locked=.locked
     fi
-    opam install --deps-only "$INFER_ROOT"/opam/infer.opam$locked &&
+    opam install --deps-only "$INFER_ROOT"/opam/infer.opam$locked "$INFER_OPAM_INSTALL_DEPS_OPTIONS" &&
     if [ -n "$SANDCASTLE" ]; then
         opam pin list | grep yojson || opam pin add yojson "${DEPENDENCIES_DIR}/yojson-1.7.0fix"
     fi
```
Along with Formula change to `ENV["INFER_OPAM_INSTALL_DEPS_OPTIONS"] = "--no-depexts"`